vllm/v1/core/kv_cache_manager.py

```diff
@@ -132,6 +135,10 @@ def append_slots(
             num_new_blocks = min(
                 num_new_blocks + self.num_preallocate_blocks,
                 self.free_block_queue.num_free_blocks,
+                # Should not exceed the maximum number of blocks per request.
+                # This is especially because the block table has the shape
+                # [..., max_num_blocks_per_req].
+                self.max_num_blocks_per_req - len(req_blocks),
```

Comment:
Add an assertion to make sure the num blocks is always >0?

Reply:
Good idea. Added.